tests/integration/mixer/outboundtrafficpolicy/helper.go

```diff
@@ -195,110 +223,51 @@ func createGateway(t *testing.T, appsNamespace namespace.Instance, serviceNamesp
 // TODO support native environment for registry only/gateway. Blocked by #13177 because the listeners for native use static
 // routes and this test relies on the dynamic routes sent through pilot to allow external traffic.
 
-// Expected is a map of protocol -> expected response codes
-func RunExternalRequestTest(expected map[string][]string, t *testing.T) {
+func RunExternalRequest(prometheus prometheus.Instance, mode TrafficPolicy, expected map[string]Response, t *testing.T) {
+	var cases = []struct {
+		name     string
+		portName string
+		host     string
+		gateway  bool
+		scheme   scheme.Instance
+	}{
+		{
+			name:     "HTTP Traffic",
```

Comment:
Overall setup client (namespace X) ---X---> destination (namespace X) restricted via Sidecar Scope
client (namespace X) ----> some-external-site.com

portName "http" case:
client (namespace X) ---HTTP request (Host: destination.<namespace-X>----> Hits listener 0.0.0.0_80 (Allow_any) cluster
Metric is istio_requests_total i.e. HTTP

Portname "https" Case:
client (namespace X) ---HTTP request (Host: destination.<namespace-X>----> Hits listener 0.0.0.0_443 (Allow_any)
Metric is istio_tcp_connections_closed_total i.e. TCP

portName "http_egress"
client (namespace X) ---HTTP request (Host: some-external-site.com----> Hits listener 0.0.0.0_80 -> VS Routing (Add Egress Header) --> Egress Gateway --> destination
Metric is istio_requests_total i.e. HTTP with destination as destination

portName TCP
client (namespace X) ---TCP request at port 9090----> Hits listener 0.0.0.0_9090 -> 0.0.0.0_150001 -> ALLOW_ANY
Metric is istio_tcp_connections_closed_total i.e. TCP